pkg/apis/metal3/v1alpha1/baremetalhost_types.go

```diff
@@ -126,6 +127,39 @@ type BMCDetails struct {
 	CredentialsName string `json:"credentialsName"`
 }
 
+// RAIDVolume defines the desired configuration of volume in RAID
+type RAIDVolume struct {
+	// Size (Integer) of the logical disk to be created in GiB.
+	// If unspecified or set be 0, the maximum capacity of disk will be used for logical disk.
```

Comment:
We should set kubebuilder:validation:Minimum=0 here to ensure we don't receive a negative value.